coderd/database/databasefake/generator.go

```diff
@@ -0,0 +1,400 @@
+				StorageMethod:  takeFirst(orig.StorageMethod, database.ProvisionerStorageMethodFile),
+				FileID:         takeFirst(orig.FileID, uuid.New()),
+				Type:           takeFirst(orig.Type, database.ProvisionerJobTypeWorkspaceBuild),
+				Input:          takeFirstBytes(orig.Input, []byte("{}")),
+				Tags:           orig.Tags,
+			})
+			require.NoError(t, err, "insert job")
+
+			output[name] = job
+
+		case database.WorkspaceResource:
+			resource, err := db.InsertWorkspaceResource(ctx, database.InsertWorkspaceResourceParams{
+				ID:         takeFirst(orig.ID, g.Lookup(name)),
+				CreatedAt:  takeFirstTime(orig.CreatedAt, time.Now()),
+				JobID:      takeFirst(orig.JobID, uuid.New()),
+				Transition: takeFirst(orig.Transition, database.WorkspaceTransitionStart),
+				// TODO: What type to put here?
+				Type: takeFirst(orig.Type, ""),
+				Name: takeFirst(orig.Name, namesgenerator.GetRandomName(1)),
+				Hide: takeFirst(orig.Hide, false),
+				Icon: takeFirst(orig.Icon, ""),
+				InstanceType: sql.NullString{
+					String: takeFirst(orig.InstanceType.String, ""),
+					Valid:  takeFirst(orig.InstanceType.Valid, false),
+				},
+				DailyCost: takeFirst(orig.DailyCost, 0),
+			})
+			require.NoError(t, err, "insert resource")
+
+			output[name] = resource
+
+		case database.File:
+			file, err := db.InsertFile(ctx, database.InsertFileParams{
+				ID:        takeFirst(orig.ID, g.Lookup(name)),
+				Hash:      takeFirst(orig.Hash, hex.EncodeToString(make([]byte, 32))),
+				CreatedAt: takeFirstTime(orig.CreatedAt, time.Now()),
+				CreatedBy: takeFirst(orig.CreatedBy, uuid.New()),
+				Mimetype:  takeFirst(orig.Mimetype, "application/x-tar"),
+				Data:      takeFirstBytes(orig.Data, []byte{}),
+			})
+			require.NoError(t, err, "insert file")
+
+			output[name] = file
+		case database.UserLink:
+			link, err := db.InsertUserLink(ctx, database.InsertUserLinkParams{
+				UserID:            takeFirst(orig.UserID, uuid.New()),
+				LoginType:         takeFirst(orig.LoginType, database.LoginTypeGithub),
+				LinkedID:          takeFirst(orig.LinkedID),
+				OAuthAccessToken:  takeFirst(orig.OAuthAccessToken, uuid.NewString()),
+				OAuthRefreshToken: takeFirst(orig.OAuthAccessToken, uuid.NewString()),
+				OAuthExpiry:       takeFirstTime(orig.OAuthExpiry, time.Now().Add(time.Hour*24)),
+			})
+
+			require.NoError(t, err, "insert link")
+
+			output[name] = link
+		default:
+			panic(fmt.Sprintf("unknown type %T", orig))
+		}
+	}
+	return output
+}
+
+func (g *Generator) Lookup(name string) uuid.UUID {
+	if name == "" {
+		// No name means the caller doesn't care about the ID.
+		return uuid.New()
+	}
+	if g.names == nil {
+		g.names = make(map[string]uuid.UUID)
+	}
+	if id, ok := g.names[name]; ok {
+		return id
+	}
+	id := uuid.New()
+	g.names[name] = id
+	return id
+}
+
+func takeFirstTime(values ...time.Time) time.Time {
+	return takeFirstF(values, func(v time.Time) bool {
+		return !v.IsZero()
+	})
+}
```

Comment:
Do we really need this?

Reply:
You are right! Didn't know time was comparable.